scripts/lighthouse/runLighthouse.js

```diff
@@ -0,0 +1,58 @@
+const lighthouse = require('lighthouse'); // eslint-disable-line import/no-extraneous-dependencies
+const chromeLauncher = require('chrome-launcher'); // eslint-disable-line import/no-extraneous-dependencies
+
+let config = {};
+
+function launchChromeAndRunLighthouse(url, opts, lConfig = null) {
+  const options = opts;
+  return chromeLauncher
+    .launch({ chromeFlags: options.chromeFlags })
+    .then(chrome => {
+      options.port = chrome.port;
+      return lighthouse(url, options, lConfig).then(results =>
+        chrome.kill().then(() => results.lhr),
+      );
+    });
+}
+
+function getScoresPerCategory(reportCategories, url) {
+  const scores = Object.keys(reportCategories).map(category => ({
+    id: reportCategories[category].id,
+    score: reportCategories[category].score,
+  }));
+
+  return { scores, url };
+}
+
+const mapScores = scores =>
+  scores.map(({ id, score }) => {
+    const expectedScore = config.thresholds[id];
+    return {
+      id,
+      score,
+      expectedScore,
+      pass: score >= expectedScore,
+    };
+  });
+
+function validateScores(resultsArray) {
+  return resultsArray.map(({ url, scores }) => ({
+    url,
+    scores: mapScores(scores),
+  }));
+}
+
+const createLighthouseRuns = configs =>
+  configs.urls.map(url =>
+    launchChromeAndRunLighthouse(url, configs.opts).then(results =>
+      getScoresPerCategory(results.categories, url),
+    ),
+  );
+
+const launchLighthouse = simorghConfig => {
+  config = simorghConfig;
```

Comment:
Maybe move this function to the top of the file? Reads a bit more nicely given that it updates a global `config` that is referred to by other functions.

Reply:
because of the order functions are invoked I'm not sure how to update this so it works that way
what I have done is change all functions to be declarations for consistency as I was using a mix of function declarations and expressions.